src/pages/InterProtocol.tsx

```diff
@@ -76,43 +80,61 @@ export function InterProtocol() {
     {},
   );
 
+  const istTokenDivisor = getTokenDivisor(boardAuxes, 'IST');
+
   const psmMinted =
-  dashboardResponse.psmMetrics.nodes.reduce((agg, node) => agg + Number(node.mintedPoolBalance), 0) / 1_000_000;
-  const psmAnchor = dashboardResponse.psmMetrics.nodes.reduce(
-    (agg, node) => agg + Number(node.anchorPoolBalance) / 10 ** (boardAuxes[node.denom] || 6),
-    0,
-  );
-  const vaultMinted =
-  dashboardResponse.vaultManagerMetrics.nodes.reduce((agg, node) => agg + Number(node.totalDebt), 0) / 1_000_000;
+    dashboardResponse?.psmMetrics?.nodes?.reduce((agg, node) => agg + Number(node?.mintedPoolBalance), 0) /
+    istTokenDivisor;
+
+  const psmAnchor = dashboardResponse.psmMetrics.nodes.reduce((agg, node) => {
+    const decimalPlaces = (boardAuxes && boardAuxes[node?.denom]) || 6;
+    const anchorPoolBalance = Number(node?.anchorPoolBalance) / 10 ** decimalPlaces;
+    return agg + anchorPoolBalance;
+  }, 0);
+
+  const vaultMinted = dashboardResponse.vaultManagerMetrics.nodes.reduce((agg, node) => {
+    const tokenDivisor = getTokenDivisor(boardAuxes, node?.liquidatingCollateralBrand);
+    return agg + Number(node?.totalDebt) / tokenDivisor;
+  }, 0);
+
   const totalMinted = psmMinted + vaultMinted;
 
   const vaultMintLimit =
-  dashboardResponse.vaultManagerGovernances.nodes.reduce((agg, node) => agg + Number(node.debtLimit), 0) / 1_000_000;
-  const psmMintLimit = dashboardResponse.psmGovernances.nodes.reduce((agg, node) => agg + Number(node.mintLimit), 0) / 1_000_000;
+    dashboardResponse?.vaultManagerGovernances?.nodes?.reduce((agg, node) => agg + Number(node?.debtLimit), 0) /
+    istTokenDivisor;
+
+  const psmMintLimit =
+    dashboardResponse?.psmGovernances?.nodes?.reduce((agg, node) => agg + Number(node?.mintLimit), 0) / istTokenDivisor;
+
   const totalMintLimit = vaultMintLimit + psmMintLimit;
 
   // bottom cards
   const totalReserve = dashboardResponse.reserveMetrics.nodes.reduce(
     (agg, node) =>
       agg +
-      node.allocations.nodes.reduce((agg_, node_) => {
+      node?.allocations?.nodes.reduce((agg_, node_) => {
+        const tokenDivisor = getTokenDivisor(boardAuxes, node_?.denom);
         const allocationInUsd =
-          ((Number(node_.value) / 1_000_000) * Number(oraclePrices[node_.denom]?.typeOutAmount || 1_000_000)) / 1_000_000;
+          ((Number(node_?.value) / tokenDivisor) * Number(oraclePrices[node_?.denom]?.typeOutAmount || tokenDivisor)) /
+          tokenDivisor;
         return agg_ + allocationInUsd;
       }, 0),
     0,
   );
   const reserveShortfall =
-  dashboardResponse.reserveMetrics.nodes.reduce((agg, node) => agg + Number(node.shortfallBalance), 0) / 1_000_000;
-  const totalLockedCollateral = dashboardResponse.vaultManagerMetrics.nodes.reduce((agg, node) => {
+    dashboardResponse?.reserveMetrics?.nodes?.reduce((agg, node) => agg + Number(node?.shortfallBalance), 0) /
+    istTokenDivisor;
```

Comment:
probably shouldnt be divided by the istTokenDivisor but rather the token divisior of each node

Reply:
Are you sure? It results in different values compared to production

Reply:
not 100%. my logic is that this value represent reserve data. when we calculate `totalReserve` above, which also represents reserve data, we also use token divisor rather than ist divisor

Reply:
I am confused too:
![image](https://github.com/agoric-labs/agoric-inter-dashboard/assets/60459172/8e4eae04-064b-495e-844d-b55ac129599a)

Reply:
![image](https://github.com/agoric-labs/agoric-inter-dashboard/assets/60459172/ce50ff59-0d42-4864-9480-9d683c32ad5d)

Reply:
i guess then istdivisor is fine